unitTests/ui/SearchInterfaceTest.ts

```diff
@@ -679,12 +679,21 @@ export function SearchInterfaceTest() {
             done();
           });
 
-          it('should allow to switch back to a waiting mode if a new empty query is performed', async done => {
+          it('should allow to switch back to a waiting mode if the first query is cancelled and a new empty query is performed', async done => {
+            $$(div).one(QueryEvents.buildingQuery, (_, args: IBuildingQueryEventArgs) => {
```

Comment:
Was this passing incorrectly before ? Why is this change needed for a seemingly unrelated UT ? 🤔

Reply:
I don't recall exactly. I believe that, yes, it was passing incorrectly. As far as I recall, the new changes I made to `Simulate` made it behave more like the real query controller, which revealed this flaw.